arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift

```diff
@@ -0,0 +1,323 @@
+        // Set the route solved by route planner to the scene.
+        // On success, enable calibration adjustment.
+        setRouteAndGraphic(route: routeResult.routes.first!) { [weak self] in
+            self?.calibrateButtonItem.isEnabled = true
+            self?.setStatus(message: "Adjust calibration before starting.")
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        arView.startTracking(.continuous, completion: nil)
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        arView.stopTracking()
+    }
+}
+
+// MARK: - Route navigation status changes
+
+extension NavigateARNavigatorViewController: AGSRouteTrackerDelegate {
+    func routeTracker(
+        _ routeTracker: AGSRouteTracker,
+        didGenerateNewVoiceGuidance voiceGuidance: AGSVoiceGuidance
+    ) {
+        setSpeakDirection(with: voiceGuidance.text)
+    }
+    
+    func routeTrackerRerouteDidStart(_ routeTracker: AGSRouteTracker) {
+        setStatus(message: "Rerouting...")
+    }
+    
+    func routeTracker(_ routeTracker: AGSRouteTracker, didUpdate trackingStatus: AGSTrackingStatus) {
+        switch trackingStatus.destinationStatus {
+        case .notReached, .approaching:
+            let currentManeuver = currentRoute.directionManeuvers[trackingStatus.currentManeuverIndex]
+            setStatus(message: currentManeuver.directionText)
+        case .reached:
+            setStatus(message: "You have arrived!")
+        default:
+            return
+        }
+    }
+    
+    func routeTracker(
+        _ routeTracker: AGSRouteTracker,
+        rerouteDidCompleteWith trackingStatus: AGSTrackingStatus?,
+        error: Error?
+    ) {
+        if let latestRoute = trackingStatus?.routeResult.routes.first {
+            if latestRoute != self.currentRoute {
+                setRouteAndGraphic(route: latestRoute)
+            }
+        }
+    }
+    
+    func setSpeakDirection(with text: String?) {
+        speechSynthesizer.stopSpeaking(at: AVSpeechBoundary.word)
+        if let text = text {
+            let speechUtterance = AVSpeechUtterance(string: text)
+            speechUtterance.rate = AVSpeechUtteranceMaximumSpeechRate * 0.5
+            speechSynthesizer.speak(speechUtterance)
+        }
+    }
+}
+
+// MARK: - AGSLocationChangeHandlerDelegate
+
+extension NavigateARNavigatorViewController: AGSLocationChangeHandlerDelegate {
+    func locationDataSource(
+        _ locationDataSource: AGSLocationDataSource,
+        locationDidChange location: AGSLocation
+    ) {
+        startButtonItem.isEnabled = true
+        // Send location updates to route tracker if actively navigating.
+        routeTracker?.trackLocation(location)
+    }
+}
+
+// MARK: - Calibration popup
+
+extension NavigateARNavigatorViewController {
+    @IBAction func showCalibrationPopup(_ sender: UIBarButtonItem) {
```

Comment:
I'm a bit confused by this method. It is titled `arcgisARView`, but it can also dismiss the popup? And popups aren't (or at least should be) dismissible by buttons outside the popup.